src/clients/c++/perf_client/load_manager.h

```diff
@@ -179,10 +179,16 @@ class LoadManager {
 
   // Holds the running status of the thread.
   struct ThreadStat {
-    ThreadStat() : status_(ni::RequestStatusCode::SUCCESS) {}
+    ThreadStat()
+        : status_(ni::RequestStatusCode::SUCCESS),
+          cb_status_(ni::RequestStatusCode::SUCCESS)
+    {
+    }
 
     // The status of the worker thread
     nic::Error status_;
+    // The status of the callback thread for async requests
+    nic::Error cb_status_;
```

Comment:
You can combine the cb_status_ into status_, then leverage the existing check for status_

Reply:
The problem with that is cb_status_ is available once the inference gets complete. For high concurrency, it is highly likely that the main thread might overwrite the erroneous status_ to success. The solution is to check status_ before allocating new values along the main thread. But there will be numerous such checks. Moreover, we would need the corresponding locking mechanism as well which might affect performance.

Reply:
Having two separate status allow us to keep the control paths separate avoiding unnecessary clobbering of status.

Reply:
Okay. But I thought that the status will never change from erroneous state to success state? i.e. we only update status_ when the status is not okay, but never the opposite.